lib/renderers/cairo_renderer.h

```diff
@@ -297,7 +297,9 @@ bm_cairo_paint(struct cairo *cairo, uint32_t width, uint32_t max_height, const s
     uint32_t border_radius = menu->border_radius;
 
     uint32_t total_item_count = menu->items.count;
-    uint32_t filtered_item_count = (menu->filter ? menu->filtered.count : total_item_count);
+    //uint32_t filtered_item_count = (menu->filter ? menu->filtered.count : total_item_count);
+    uint32_t filtered_item_count;
+    bm_menu_get_filtered_items(menu, &filtered_item_count);
```

Comment:
why was this changed?

Reply:
I'm unsure about if it was a bug with the match counter or some code that I changed, but occasionally, the filtered items wouldn't update properly, so I decided to get the filtered item count from there and it seemed to fix the bug.

Reply:
On second thought, I'm pretty sure there was an issue with the ternary statement that incorrectly set the count to 0 occasionally. I believe this was an issue with the counter itself.